core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -134,7 +134,7 @@ class InvokerReactive(
     UserEvents.send(
       producer, {
         val activation = Activation(
-          activationResult.namespace + EntityPath.PATHSEP + activationResult.name,
+          activationResult.annotations.getAs[String](WhiskActivation.pathAnnotation).getOrElse("unknown_action"),
```

Comment:
the path should always be present no?

Reply:
Ah, gotcha. I guess we should use Option's everywhere? It's the same thing in the end. Meaningless default value vs. explicitly optional field?